diff --git a/window-copy.c b/window-copy.c
index 507461d..3f0bb2b 100644
--- a/window-copy.c
+++ b/window-copy.c
@@ -761,10 +761,10 @@ window_copy_command(struct window_pane *wp, struct client *c, struct session *s,
 	} else if (args->argc == 2 && *args->argv[1] != '\0') {
 		argument = args->argv[1];
 		if (strcmp(command, "copy-pipe") == 0) {
-			if (s != NULL) {
+			if (s != NULL)
 				window_copy_copy_pipe(wp, s, NULL, argument);
-				window_pane_reset_mode(wp);
-			}
+			window_copy_clear_selection(wp);
+			window_copy_redraw_screen(wp);
 		}
 		if (strcmp(command, "goto-line") == 0)
 			window_copy_goto_line(wp, argument);
